### DISCUSSION TITLE
Support for password hashing

### DISCUSSION THREAD

**Discussion by rossholdway** [denoland/deno#21618] [Ideas]
Hello,

I see Bun has built in password hashing support for Argon2 variants and Bcrypt (https://bun.sh/docs/api/hashing).

I know there are various third-party libraries out there, but I imagine something like this is an important feature for a number of Deno users. Can we ever expect to see something similar in Deno / Deno std? Thanks!

**Comment by halvardssm** [comment]
I recently started a new project to extend Deno STD. It encourages the addition of functionality that exists in other standard libraries (PHP, Go etc.), and also allows for functionality that benefits from being in a standard library rather than in third party modules. Here is the link: https://jsr.io/@stdext/crypto

**Comment by brettchalupa** [comment]
I would love to see argon2 be included in the Deno std library to make password hashing as simple and fast as everything else in Deno is.

**Reply by brettchalupa** [reply]
I have spent more time digging into password hashing and salting with Deno and came across https://jsr.io/@felix/argon2, a well-maintained library that uses Rust’s implementation of the algorithm. Unfortunately, it relies upon FFI which can’t be deployed to Deno Deploy as far as I can tell. 

I don’t see a simple way to implement secure password hashing, which is a requirement for most web application with auth, with Deno and Deno Deploy. 

If Deno provided a std lib implementation, that’d really make it more viable to use Deno and Deno Deploy for building full stack web apps with auth.

**Reply by brettchalupa** [reply]
For now I'm using https://jsr.io/@rabbit-company/argon2id which uses a WASM build of argon2 and is working on Deno Deploy. But it'd still be nice to have password hashing without FFI and without WASM.

**Reply by KnorpelSenf** [reply]
I suggest you take a look at https://jsr.io/@noble/hashes#argon2

**Comment by rossholdway** [comment]
I opened feature request https://github.com/denoland/std/issues/5899 awhile back to help track this.
